common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -227,9 +239,117 @@ public boolean onTouch(final View view, final MotionEvent event) {
         mWebView.getSettings().setBuiltInZoomControls(webViewZoomControlsEnabled);
         mWebView.getSettings().setSupportZoom(webViewZoomEnabled);
         mWebView.setVisibility(View.INVISIBLE);
+        mWebView.getSettings().setPluginState(WebSettings.PluginState.ON);
         mWebView.setWebViewClient(webViewClient);
+        mWebView.setWebChromeClient(new WebChromeClient() {
+            @Override
+            public void onPermissionRequest(final PermissionRequest request) {
+                // We can only grant or deny permissions for video capture/camera.
+                // To avoid unintentionally granting requests for not defined permissions.
+                if (iSPermissionRequestForCamera(request)) {
+                    Logger.info(methodTag, "Camera request received.");
+                    mCameraPermissionRequest = request;
+                    if (isCameraPermissionGranted()) {
+                        Logger.info(methodTag, "Camera permission already granted.");
+                        acceptCameraRequest();
+                    } else if (shouldShowRequestPermissionRationale(Manifest.permission.CAMERA)) {
+                        Logger.info(methodTag, "Show camera rationale.");
+                        showCameraRationale();
+                    } else {
+                        requestCameraPermissionFromUser();
+                    }
+                }
+            }
+        });
+    }
+
+    /**
+     * Call this method to grant the permission to access the camera resource.
+     * The granted permission is only valid for the current WebView.
```

Comment:
What does `current` mean here? On next launch of WebView will the user need to grant permission again?

Reply:
That will depend on the behavior the user has chosen; the user can approve the camera permission just for this request or for this and all future requests.
![image](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/assets/76129899/e4a63008-b2c4-4f16-85f4-c51c67af914d)